app/forms/openid_connect_authorize_form.rb

```diff
@@ -68,6 +66,22 @@ def service_provider
     @_service_provider ||= ServiceProvider.from_issuer(client_id)
   end
 
+  def link_identity_to_client_id(current_user, rails_session_id)
```

Comment:
Ooh or actually the `identity` itself is a link between a user and service provider.

So maybe: `link_user_to_service_provider` ?